### PR TITLE
test: confidence net before CVE/security fixes

### PR DESCRIPTION
**Pull request by YoanRoullard**
## Summary
- Adds 3 characterization tests so we can bump dependencies (CVE / security fixes) with a regression net.
- No source code changes — only `tests/` and `docs/superpowers/` are touched.
- Suite goes from 20 to 23 OKs, lint clean.

## What's covered

| Layer | Test | What it pins |
|---|---|---|
| Unit (`notes-placement`) | fallback when `activeNoteId` is not found | The implicit `\|\| notes[0]` anchor — easy to break in a refactor. |
| Unit (`notes-placement`) | `offsetY = 0` for a single note | Freezes the current behavior around the latent `\|\| offsetY` short-circuit. Inline comment warns future maintainers not to remove it. |
| Integration (`sidenotes-wrapper`) | `@gutter` is propagated end-to-end | No existing integration test exercised the gutter arg through the wrapper. |

## Explicitly out of scope
- No `ember-cli-code-coverage` (avoided adding a dep right before bumping deps).
- No source fixes (notably the `\|\| offsetY` short-circuit is documented as fragile but **not** fixed here — separate PR after CVE work).
- No new edge cases beyond the 3 above.

## Test plan
- [ ] `npm run test:ember` → 23 OKs, 0 fails
- [ ] `npm run lint` → green
- [ ] CI passes

### CODE REVIEW

**Comment by thiberaw** [comment]
**question (non-blocking) on `docs/superpowers/`:**

This PR establishes a brand-new `docs/superpowers/{plans,specs}/` tree. These docs are AI-workflow artifacts (they reference `superpowers:subagent-driven-development`, contain implementation checklists with `- [ ]` gates, and quote pre-merge test counts like "19 functional tests → 23 OKs" that go stale the moment they land).

Worth deciding at the repo level before we set a precedent:

1. **Commit them** as living history of "why this test exists" — keep, but strip the transient bits (counts, gate steps, absolute paths).
2. **Don't commit them** — add `docs/superpowers/` to `.gitignore` so future agentic sessions don't accidentally land them.
3. **Move them** to a personal scratch (e.g. a notes repo or a gist linked from the PR description).

The decision affects every future PR that uses superpowers, not just this one. Could you confirm with the team which way we want to go?

File: docs/superpowers/plans/2026-05-13-confidence-tests-before-cve-fix.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+Run:
+```bash
+ls /home/yoan/_dev/ember-sidenotes/node_modules >/dev/null 2>&1 && echo OK || echo MISSING
+```
+
+If `MISSING`, run:
+```bash
+cd /home/yoan/_dev/ember-sidenotes && npm install
+```
```
Comment (thiberaw) [inline docs/superpowers/plans/2026-05-13-confidence-tests-before-cve-fix.md]: **nitpick:** The plan doc hard-codes `/home/yoan/_dev/ember-sidenotes` in ~7 shell snippets (lines 36, 41, 50, 102, 145, 215, 234, 243, 252, 259, 268). If this file is kept in the repo (see top-level comment), please strip the absolute path — anyone else replaying the plan would have a different home dir, and we'd rather not advertise contributor personal paths. A `cd $REPO_ROOT && …` or just relative `npm run …` from the repo root would do.

**Comment by thiberaw** [comment]
**praise:** Good test selection. The `notes.find(...) || notes[0]` fallback and the `@gutter` end-to-end propagation were both genuine gaps the existing suite didn't cover, and the comment on `it preserves offsetY = 0` is honest about pinning current behavior rather than asserting correctness — exactly what a regression net should do before a dep bump. Verified the three new tests against `addon/utils/notes-placement.js` and the wrapper's `placeNotes` path; expected values check out.

One observation, no action needed: the `it preserves offsetY = 0` test would still pass if the latent `|| offsetY` short-circuit were fixed properly (e.g., `Math.min(prevNote?.top ?? offsetY, offsetY)`), because the single-note case bottoms out at `0` either way. Fine for "freeze current behavior", but worth knowing it's not a *tight* discriminator for the bug — the follow-up bug-fix PR will want a multi-note case to prove the new code path.

The non-test changes (`docs/superpowers/`) are flagged separately above.

**Comment by thiberaw** [comment]
## CI investigation — red checks are pre-existing, unrelated to this PR

**TL;DR:** Of the 7 "failing" checks in the UI, only **1 is a real failure** (`Try: ember-release`) and the other 6 are **CANCELLED** by `fail-fast: true`. The same crash reproduces on `master` (latest run [#26141425037](https://github.com/concordnow/ember-sidenotes/actions/runs/26141425037), 2026-05-20). This PR adds only `tests/*` and `docs/*` — it cannot have caused it.

The `Tests` job (Node 16 + pinned Ember 3.28) passes in 51s, which is the meaningful signal for this PR.

### Root cause

The workflow pins `NODE_VERSION: '16'` (`.github/workflows/main.yml:12`), but the `ember-release` ember-try scenario installs the **latest** `ember-source` + `ember-cli`, which has dropped Node 16. CI explicitly warns:

> `WARNING: Node v16.20.2 is not tested against Ember CLI on your platform.`

…then crashes during vendor-file init:

```
TypeError: Cannot read properties of undefined (reading 'debug')
    at EmberAddon._initVendorFiles (node_modules/ember-cli/lib/broccoli/ember-app.js:413:38)
    at new EmberApp           (node_modules/ember-cli/lib/broccoli/ember-app.js:115:10)
    at new EmberAddon         (node_modules/ember-cli/lib/broccoli/ember-addon.js:61:5)
    at module.exports         (ember-cli-build.js:6:13)
```

With `fail-fast: true` (`.github/workflows/main.yml:58`), the other 6 matrix jobs (`ember-classic`, `ember-lts-3.20/3.24`, `embroider-safe/optimized`, `ember-default-with-jquery`) are cancelled the moment `ember-release` dies. They never ran to completion.

### Suggested fix (separate PR / issue)

In order of effort:

1. **Bump `NODE_VERSION` to `'18'` or `'20'`** in `.github/workflows/main.yml:12`. Cheapest fix.
2. **Pin `ember-release`** to a known-working channel snapshot in `config/ember-try.js`.
3. **Drop the `ember-release` scenario** — this addon pins peers around 3.20/3.24/3.28, so the bleeding edge has limited value here.
4. **Set `fail-fast: false`** in `.github/workflows/main.yml:58` so a single regression doesn't mask the other 6 scenarios. Defensive, complements any of the above.

Not blocking this PR — flagging so the red checks don't slow merge.

**Comment by YoanRoullard** [comment]
@thiberaw j'ai retiré docs/superpowers

**Review by thiberaw — APPROVED** [review]
LGTM. Re-reviewed after `docs/superpowers/` removal (`e72aa1a`) — remaining diff is the three characterization tests + the `.gitignore` rule.

Traced all three tests against `addon/utils/notes-placement.js` and the wrapper's `placeNotes` path; expected values check out. The integration `@gutter` test is genuine new coverage (existing gutter test only exercises the util layer).

Outstanding items are correctly scoped to follow-up PRs:
- Bug-fix PR should add a multi-note case for the `|| offsetY` short-circuit (the single-note pin passes by coincidence — already noted in the inline code comment).
- CI matrix Node 16 / `ember-release` crash is pre-existing on master; needs its own PR.

Good regression net before the CVE bumps. Ship it.
